src/__tests__/hooks.test.js

```diff
@@ -654,6 +654,78 @@ describe('hooks', () => {
     expect(innerRenderCount).toEqual(1);
   });
 
+  it('does not cause unnecessary dispatches due to object literals passed to useTracking', () => {
+    let dispatchCount = 0;
+
+    const App = () => {
+      const [state, setState] = React.useState({});
+
+      useTracking(
+        {},
+        {
+          dispatchOnMount: () => {
+            dispatchCount += 1;
+          },
+        }
+      );
+
+      return (
+        <div className="App">
+          <h1>
+            Extra dispatches caused by new object literals passed on re-render
+          </h1>
+          <button
+            onClick={() => setState({ count: state.count + 1 })}
+            type="button"
+          >
+            Update trackingData and options objects
+          </button>
+        </div>
+      );
+    };
+
+    const wrapper = mount(<App />);
+
+    wrapper.find('button').simulate('click');
```

Comment:
Maybe click it a couple times and wrap in `act()` for good measure? Would be great to also assert that the state is actually updated to confirm.

Reply:
Great call on this. Adding a couple extra clicks exposed a bug. I decided not to wrap the clicks in `act()` because it wasn't making a difference in the result, and it turns out that's because as of React 16.8, if we use `mount`, enzyme [wraps calls](https://enzymejs.github.io/enzyme/#reacttestutilsact-wrap) to `.simulate` in `act` for us. I think we just needed it in the other spot to handle the asynchronicity.

I still want to do some manual testing again before we merge this just to be sure everything is working properly. Should we aim for a Monday merge?